docs: address DA comments (#4641)
AztecProtocol/aztec-packages: yellow-paper/docs/data-publication-and-availability/index.md

But before we dive into that, let us take a higher level look at Aztec to get a understanding of our requirements.

In particular, we will be looking at what is required to give observers (nodes) different guarantees similar to what Jon did in [his post](https://dba.xyz/do-rollups-inherit-security/).
This can be useful to get an idea around what we can do for data publication and availability later.

## Rollup 101

<!-- Rename section to convey exactly what it's about (what are we catching up on?). -->

A rollup is broadly speaking a blockchain that put its blocks on some other chain (the host) to make them available to its nodes.
Most rollups have a contract on this host blockchain which validates its state transitions (through fault proofs or validity proofs) taking the role of a full-validating light-node, increasing the accessibility of running a node on the rollup chain, making any host chain node indirectly validate its state.

With its state being validated by the host chain, the security properties can eventually be enforced by the host-chain if the rollup chain itself is not progressing.
Bluntly, the rollup is renting security from the host.

> The essential difference between a L1 and a rollup then comes down to who are required for block production (liveness) and to convince the validating light-node (security), for the L1 it is the nodes of the L1, and for the Rollup the nodes of its host (eventually).

Reviewer: ```suggestion
The essential difference between an L1 and a rollup then comes down to who are required for block production (liveness) and to convince the validating light-node (security).
For the L1 it is the nodes of the L1, and for the Rollup the nodes of its host (eventually).
```

it's easier to read text when the sentences are not giga long.

### Consensus

For a consensus based rollup you can run LMD-Ghost similarly to Ethereum, new blocks are built like Ethereum, and then eventually reach the host chain where the light client should also validate the consensus rules before progressing state.
In this world, you have a probability of re-orgs trending down as blocks are built upon while getting closer to the finalization.
Users can then rely on their own confirmation rules to decide when they deem their transaction confirmed.
You could say that the transactions are pre-confirmed until they convince the validating light-client on the host.

### No-consensus

If there is no explicit consensus for the Rollup, staking can still be utilized for leader selection, picking a distinct sequencer which will have a period to propose a block and convince the validating light-client.
The user can as earlier define his own confirmation rules and could decide that if the sequencer acknowledge his transaction, then he sees it as confirmed.

> This can be done fully on trust, of with some signed message the user could take to the host and "slash" the sequencer for not upholding his part of the deal.

Reviewer: This sentence is weird.

:::info Fernet
Fernet lives in this category if you have a single sequencer active from the proposal to proof inclusion stage.
:::

Common for both consensus and no-consensus rollups is that the user can decide when he deems his transaction confirmed.
If the user is not satisfied with the guarantee provided by the sequencer, he can always wait for the block to be included in the host chain and get the guarantee from the host chain consensus rules.

## Data Availability and Publication

As alluded to earlier, we belong to the school of thought that Data Availability and Publication is different things.
Generally, what is often referred to as Data Availability is merely Data Publication, e.g., whether or not the data have been published somewhere.

> For data published on Ethereum you will currently have no issues getting a hold of the data because there are many full nodes and they behave nicely, but they could are not guaranteed to be so.

Reviewer: ```suggestion
For data published on Ethereum you will currently have no issues getting a hold of the data because there are many full nodes and they behave nicely, but they are not guaranteed to continue doing so.
```

New nodes are essentially bootstrapped by other friendly nodes.

With that out the way, it would be prudent to elaborate on our definition from earlier:

- **Data Availability**: The data is available to anyone right now
- **Data Publication**: The data was available for a period when it was published.

With this split, we can map the methods of which we can include data for our rollup.
Below we have included only systems that are live or close to live where we have good ideas around the throughput and latency of the data.

> The latency is based on using Ethereum L1 as the home of the validating light node, and will therefore be the latency from being included in the data layer until statements can be included in the host chain.

Reviewer: ```suggestion
The latency is based on using Ethereum L1 as the home of the validating light node, and will therefore be the latency between point in time when data is included on the data layer until a point when statements about the data can be included in the host chain.
```

Found the original sentence hard to understand.

### Data Layer outside host

When using a data layer that is not the host chain, cost (and safety guarantees) are reduced, and we rely on some "bridge" to tell the host chain about the data.

> This must happen before our validating light node can progress the block, hence the block must be published, and the host must know about it before the host can use it as input to block validation.

Reviewer: ```suggestion
This must happen before our validating light node can progress the block.
Therefore the block must be published, and the host must know about it before the host can use it as input to block validation.
```

#### Celestia

Celestia mainnet is starting with a limit of 2 mb/block with 12 second blocks supporting ~166 KB/s.

> But they are working on increasing this to 8 mb/block.

Reviewer: ```suggestion
:::note
They are working on increasing this to 8 mb/block.
:::
```

Think this will be nicer as a note.

As Celestia has just recently launched, it is unclear how much competition there will be for the data throughput, and thereby how much we could expect to get a hold of.

> Since the security assumptions differ greatly from the host chain (Ethereum) few L2's have been built on top of it yet, and the demand is to be gauged in the future.

Reviewer: ```suggestion
Since the security assumptions differ greatly from the host chain (Ethereum) few L2s have been built on top of it yet, and the demand is to be gauged in the future.
```

> Beyond the pure data throughput, we also need Ethereum L1 to know that the data was made available on Celestia, which here will require the [blobstream](https://blog.celestia.org/introducing-blobstream/) (formerly the quantum gravity bridge) to relay data roots that the rollup contract can process.This is currently done approximately every 100 minutes.

Reviewer: ```suggestion
Beyond the pure data throughput, we also need Ethereum L1 to know that the data was made available on Celestia, which here will require the [blobstream](https://blog.celestia.org/introducing-blobstream/) (formerly the quantum gravity bridge) to relay data roots that the rollup contract can process.
This is currently done approximately every 100 minutes.
```

- L1 -> L2
- L2 -> L1

Some of these can be moved around between layers, and others are hard-linked to live on the host.
For one, moving the cross-chain message L1 -> L2 and L2 -> L1 anywhere else than the host is fighting an up-hill battle.
Also, beware that the state for L2 -> L1 messages is split between the data layers, as the messages don't strictly need to be available from the L2 itself, but must be for consumption on L1.

We need to know what these things are to be able to progress the state.
Without having the state, we don't know how the output of a state transition should look and cannot prove it.

> Beyond the above data that is important to everyone, we also have data that is important to _someone_, these are the logs, both unencrypted and encrypted.

Reviewer: ```suggestion
Beyond the above data that is important to everyone, we also have data that is important to _someone_.
These are encrypted and unencrypted logs.
```

> Knowing the historic logs are not required to progress the chain, but are important for the users to ensure that they learn about their notes etc.

Reviewer: ```suggestion
Knowing the historic logs is not required to progress the chain, but they are important for the users to ensure that they learn about their notes etc.
```
